### PR TITLE
Document consuming CDC data with Pulsar and Kinesis

### LINKED ISSUE #652: Document the support of using Pulsar and Kinesis to transfer CDC data
### Related code PR

_No response_

### Which part(s) of the docs might be affected or should be updated? And how?

We mention in the docs that users can use the Kafka connector to convert CDC data to Kafka topics. Actually, both Pulsar and Kinesis can also be used for the same purpose. 
For Pulsar, this connector from Pulsar is needed (on top of the Pulsar connector in RisingWave): https://pulsar.apache.org/docs/2.11.x/io-cdc/
For Kinesis, AWS DMS is needed: https://aws.amazon.com/blogs/big-data/stream-change-data-to-amazon-kinesis-data-streams-with-aws-dms/

The connector settings for transferring CDC data are the same as non-CDC data. The only difference is ROW FORMAT being various JSON formats (Debezium JSON, Maxwell, Canal JSON, etc). Additionally, as ROW FORMAT is various JSON formats, MESSAGE and SCHEMA LOCATION are not needed. 

@tabVersion Please confirm my above description is accurate. Thanks. 

To accommodate the changes, we need to change the title of "CDC via Kafka" to "CDC via message brokers", and fit in the descriptions. 

We'll also need to make changes accordingly in "Ingest data from MySQL" and "Ingest CDC data from PostgreSQL", to say that people can use these three message brokers as intermediate medium to transfer CDC data from databases to RisingWave.

### Reference

_No response_

### PR DESCRIPTION
**Pull request by emile-00**
## Info
- **Description**: 
Edit MySQL and PostgreSQL CDC guides to include instructions on consuming CDC data with Pulsar and Kinesis
Change Ingest CDC via Kafka guide to Ingest CDC via message brokers guide

- **Preview**: 
[ Paste the preview link to the edited page here. ]

- **Related code PR**: 
[ Provide the link to the code PR here. For example, https://github.com/risingwavelabs/risingwave/pull/4085. Delete this part if there's no code PR related. ]

- **Related doc issue**: 
Resolves https://github.com/risingwavelabs/risingwave-docs/issues/652

- **Notes**: 
[ Any additional information? ]

## Before merging
  - [ ] (For version-specific PR) I have selected the corresponding software version in **Milestone** and linked the related doc issue to this PR in **Development**.
  - [ ] I have acquired the approval from the owner (and optionally the reviewers) of the code PR and at least one tech writer (`bernscode`, `CharlieSYH`, `emile-00`, & `hengm3467`). 
  - [ ] I have checked the doc site preview, and the updated parts look good. <details><summary>How?</summary>Scroll down and open this link: image
</details>

### CODE REVIEW

File: docs/create-source/create-source-cdc.md [hunk 1]
```diff
@@ -1,33 +1,41 @@
 ---
 id: create-source-cdc
 title: CDC via Kafka
-description: Ingest CDC data via Kafka.
+description: Ingest CDC data via a message broker.
```
Comment (tabVersion) [inline docs/create-source/create-source-cdc.md]: from a message queue?
message broker means the machine where message queue service is located.

File: docs/create-source/create-source-cdc.md [hunk 2]
```diff
@@ -1,33 +1,41 @@
...
 
 RisingWave provides native MySQL and PostgreSQL CDC connectors. With these CDC connectors, you can ingest CDC data from these databases directly, without setting up additional services like Kafka.
 
-If Kafka is part of your technical stack, you can also use the Kafka connector in RisingWave to ingest CDC data in the form of Kafka topics from databases into RisingWave. You need to use a CDC tool such as [Debezium connector for MySQL](https://debezium.io/documentation/reference/stable/connectors/mysql.html), [Maxwell's daemon](https://maxwells-daemon.io/), or [TiCDC](https://docs.pingcap.com/tidb/dev/ticdc-overview) to convert CDC data into Kafka topics.
+If Kafka is part of your technical stack, you can connect Kafka to RisingWave to ingest CDC data in the form of Kafka topics from databases into RisingWave. You need to use a CDC tool such as [Debezium connector for MySQL](https://debezium.io/documentation/reference/stable/connectors/mysql.html), [Maxwell's daemon](https://maxwells-daemon.io/), or [TiCDC](https://docs.pingcap.com/tidb/dev/ticdc-overview) to convert CDC data into Kafka topics.
```
Comment (tabVersion) [inline docs/create-source/create-source-cdc.md]: to produce CDC messages and fill them into Kafka topics

we are not converting CDC data into topics, right?

File: docs/create-source/create-source-cdc.md [hunk 3]
```diff
@@ -1,33 +1,41 @@
...
 
-This topic describes the configurations for using the Kafka connector in RisingWave to ingest CDC data. For complete step-to-step guides about using the native CDC connector to ingest MySQL and PostgreSQL data, see [Ingest data from MySQL](../guides/ingest-from-mysql-cdc.md) and [Ingest data from PostgreSQL](../guides/ingest-from-postgres-cdc.md). For completeness, instructions about using additional CDC tools and the Kafka connector to ingest CDC data are also included in these two topics.
+If Pulsar is part of your technical stack, you can connect Pulsar to RisingWave to ingest CDC data in the form of Pulsar topics from databases into RisingWave. You need to use a CDC tool such as the [Debezium connector for MySQL](https://debezium.io/documentation/reference/stable/connectors/mysql.html), or the [Canal source connector](https://pulsar.apache.org/docs/2.11.x/io-canal-source/) to convert CDC data into Pulsar topics.
```
Comment (tabVersion) [inline docs/create-source/create-source-cdc.md]: ditto

File: docs/create-source/create-source-cdc.md (hunk 2 above)
Comment (emile-00) [reply docs/create-source/create-source-cdc.md]: Thanks for catching that incorrect wording!

File: sidebars.js [hunk 4]
```diff
@@ -201,7 +201,7 @@ const sidebars = {
         },
           {
           type: 'doc',
-          label: 'CDC via Kafka',
+          label: 'CDC via message queues',
```
Comment (hengm3467) [inline sidebars.js]: @emile-00 Let's use "event streaming systems" instead. Neither of them (Kafka, Pulsar and Kinesis) wants to be viewed as a typical message queue.

File: docs/create-source/create-source-cdc.md [hunk 5]
```diff
@@ -1,33 +1,41 @@
...
 
 RisingWave provides native MySQL and PostgreSQL CDC connectors. With these CDC connectors, you can ingest CDC data from these databases directly, without setting up additional services like Kafka.
 
-If Kafka is part of your technical stack, you can also use the Kafka connector in RisingWave to ingest CDC data in the form of Kafka topics from databases into RisingWave. You need to use a CDC tool such as [Debezium connector for MySQL](https://debezium.io/documentation/reference/stable/connectors/mysql.html), [Maxwell's daemon](https://maxwells-daemon.io/), or [TiCDC](https://docs.pingcap.com/tidb/dev/ticdc-overview) to convert CDC data into Kafka topics.
+If Kafka is part of your technical stack, you can connect Kafka to RisingWave to ingest CDC data in the form of Kafka topics from databases into RisingWave. You need to use a CDC tool such as [Debezium connector for MySQL](https://debezium.io/documentation/reference/stable/connectors/mysql.html), [Maxwell's daemon](https://maxwells-daemon.io/), or [TiCDC](https://docs.pingcap.com/tidb/dev/ticdc-overview) to produce CDC messages and send them to Kafka topics.
```
Comment (hengm3467) [inline docs/create-source/create-source-cdc.md]: Have three paragraphs with similar wording is a bit verbose. And I find the starting sentence is a bit misleading (me to blame). Even if Kafka is already part of their stack, they may still choose the native CDC connector (one less step). We can say something like: 

You can use event streaming systems like Kafka, Pular, or Kinesis to stream changes from MySQL, PostgreSQL, and TiDB to RisingWave. In this case, you need additional CDC tools to stream the changes, and specify the corresponding formats when ingesting the streams into RisingWave.

For the supported data formats, see [may need to create a separate section for the supported data formats below]().

File: docs/create-source/create-source-cdc.md [hunk 6]
```diff
@@ -1,33 +1,41 @@
...
+
+If Kinesis is part of your technical stack, you can connect Kinesis to RisingWave to ingest CDC data in the form of Kinesis data streams from databases into RisingWave. You need to use a CDC tool such as [Maxwell's daemon](https://maxwells-daemon.io/), [AWS DMS](https://aws.amazon.com/blogs/big-data/stream-change-data-to-amazon-kinesis-data-streams-with-aws-dms/), or the [Debezium embedded engine](https://debezium.io/blog/2018/08/30/streaming-mysql-data-changes-into-kinesis/) to produce CDC messages and send them to Kinesis data streams.
+
+This topic describes the configurations for using RisingWave to ingest CDC data from a message broker. For complete step-to-step guides about using the native CDC connector to ingest MySQL and PostgreSQL data, see [Ingest data from MySQL](../guides/ingest-from-mysql-cdc.md) and [Ingest data from PostgreSQL](../guides/ingest-from-postgres-cdc.md). For completeness, instructions about using additional CDC tools and a message broker to ingest CDC data are also included in these two topics.
```
Comment (hengm3467) [inline docs/create-source/create-source-cdc.md]: "message broker" -> "event streaming system"

File: docs/create-source/create-source-cdc.md [hunk 7]
```diff
@@ -1,33 +1,41 @@
...
+   For Debezium JSON, you can use the [Debezium connector for MySQL](https://debezium.io/documentation/reference/stable/connectors/mysql.html) or [Debezium connector for PostgreSQL](https://debezium.io/documentation/reference/stable/connectors/postgresql.html) to convert CDC data to Kafka or Pulsar topics, or Kinesis data streams.
 
 - Maxwell JSON (for MySQL only)
 
-  For Maxwell JSON (`ROW FORMAT MAXWELL`), you need to use [Maxwell's daemon](https://maxwells-daemon.io/) to convert MySQL data changes to Kafka topics. To learn about how to configure MySQL and deploy Maxwell's daemon, see the [Quick Start](https://maxwells-daemon.io/quickstart/).
+  For Maxwell JSON (`ROW FORMAT MAXWELL`), you need to use [Maxwell's daemon](https://maxwells-daemon.io/) to convert MySQL data changes to Kafka or Pulsar topics. To learn about how to configure MySQL and deploy Maxwell's daemon, see the [Quick Start](https://maxwells-daemon.io/quickstart/).
```
Comment (hengm3467) [inline docs/create-source/create-source-cdc.md]: Maxwell supports Kafka and Kinesis, but not Pulsar according to their docs: https://maxwells-daemon.io/quickstart/

File: docs/guides/ingest-from-mysql-cdc.md [hunk 8]
```diff
@@ -21,12 +21,11 @@ You can ingest CDC data from MySQL in two ways:
 
   This connector is included in RisingWave. With this connector, RisingWave can connect to MySQL directly to obtain data from the binlog without starting additional services.
 
-- Using a CDC tool and the Kafka connector
+- Using a CDC tool and a message broker
```
Comment (hengm3467) [inline docs/guides/ingest-from-mysql-cdc.md:24]: a message broker -> an event streaming system

File: docs/guides/ingest-from-mysql-cdc.md [hunk 9]
```diff
@@ -304,22 +303,24 @@ CREATE TABLE orders (
 );
 ```
 
-## Using a CDC tool and the Kafka connector
+## Use a CDC tool and a message broker
```
Comment (hengm3467) [inline docs/guides/ingest-from-mysql-cdc.md]: Same as above.

File: docs/guides/ingest-from-postgres-cdc.md [hunk 10]
```diff
@@ -17,11 +17,11 @@ You can ingest CDC data from PostgreSQL in two ways:
 
     This connector is included in RisingWave. With this connector, RisingWave can connect to PostgreSQL directly to obtain data from the binlog without starting additional services.
 
-- Using a CDC tool and the Kafka connector
+- Using a CDC tool and a message broker
```
Comment (hengm3467) [inline docs/guides/ingest-from-postgres-cdc.md:20]: Same as one of the above comments.

File: docs/guides/ingest-from-postgres-cdc.md [hunk 11]
```diff
@@ -280,34 +279,99 @@ export const svg = rr.Diagram(
  ```
 
 
-## Use the Debezium connector for PostgreSQL
+## Use a CDC tool and a message broker
```
Comment (hengm3467) [inline docs/guides/ingest-from-postgres-cdc.md]: Same here.

File: docs/guides/ingest-from-mysql-cdc.md [hunk 12]
```diff
@@ -304,22 +303,24 @@ CREATE TABLE orders (
...
 
 <Tabs>
-<TabItem value="Debezium connector for MySQL" label="Debezium connector for MySQL" default>
+<TabItem value="Kafka" label="Kafka" default>
 
-### Set up MySQL
+### Option 1: Deploy the Debezium connector
```
Comment (hengm3467) [inline docs/guides/ingest-from-mysql-cdc.md]: Now I feel we may have made this topic over-complicated. Can we evaluate the possibility of leaving out all the CDC tool related content, and let users figure out on their own. The scenario of using these event streaming systems in between takes up only a fraction, if any, of the cases. And we are spending a much bigger share of our time on it.

Comment (emile-00) [reply docs/guides/ingest-from-mysql-cdc.md]: I've removed the connector content from the MySQL and PG guides and updated the structure of the guide. Please take a look and let me know if additional changes need to be made.

**Review by hengm3467 — APPROVED** [review]
LGTM. Thanks!

File: docs/guides/ingest-from-mysql-cdc.md (hunk 12 above)
Comment (hengm3467) [reply docs/guides/ingest-from-mysql-cdc.md]: @emile-00 All good now. Please merge this PR when you feel ready.

**Pull request merged**